rust/kernel/clk.rs

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Clock subsystem
+//!
+//! C header: [`include/linux/clk.h`](../../../../include/linux/clk.h)
+
+use crate::{
+    bindings,
+    device::Device,
+    error::{from_kernel_err_ptr, Error, Result},
+    str::CStr,
+};
+
+/// Represents `struct clk *`
+pub struct Clk(*mut bindings::clk);
+
+impl Clk {
+    /// returns value of rate field of `struct clk`
+    pub fn get_rate(&self) -> u32 {
+        // SAFETY: self object contains aquired, valid pointer to `struct clk`
+        unsafe { bindings::clk_get_rate((*self).0) }
+    }
+}
+
+/// clk_prepare_enable helps cases using clk_enable in non-atomic context
+pub fn prepare_enable(clk: &mut Clk) -> Result {
```

Comment:
Should this be a method of `Clk`?

Reply:
Yes, I think it can. It is being strongly associated with a type.
The common clock functions (`enable()`, `prepare()`, getters/setters, etc.) will suite as methods well.

Reply:
Sounds good to me.